Automattic/jetpack
# Migrate the vimeo shortcode AMP logic to Jetpack from the AMP plugin

#### Summary 

Proposes migrating the AMP conversion of the Jetpack `[vimeo]` shortcode from the [AMP plugin](https://github.com/ampproject/amp-wp) to Jetpack.

Fixes https://github.com/ampproject/amp-wp/issues/3309

#### Changes proposed in this Pull Request:

Would you be open to migrating the [AMP plugin's conversion](https://github.com/ampproject/amp-wp/blob/ea9e6fb9d262e699ea64978794840ba9868715f6/includes/embeds/class-amp-vimeo-embed.php#L71) of Jetpack's `[vimeo]` shortcode into Jetpack?

This should give you more control over how your shortcode works.

As Weston mentioned in https://github.com/ampproject/amp-wp/issues/3309#issuecomment-533255590, there's been [ongoing work](https://github.com/ampproject/amp-wp/issues/1021) to move Jetpack functionality from the AMP plugin into Jetpack.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
This is a new feature to Jetpack, but mainly a migration of the AMP plugin's logic into Jetpack. 

#### Testing instructions:
1. Ensure that `wp-config.php` has `define( 'JETPACK_DEV_DEBUG', true);`
2. In `/wp-admin`, in the Jetpack 'Settings' page, and in the 'Writing' tab, ensure this is toggled on:

<img width="818" alt="jetpack-here" src="https://user-images.githubusercontent.com/4063887/67992321-dded6400-fc01-11e9-81ea-9a05716d1eca.png">

3. Fetch this PR's branch
4. Clone the AMP plugin 
```
$ git clone --recursive https://github.com/ampproject/amp-wp.git amp && cd amp
```
5. Fetch this PR of the AMP plugin, which deactivates the AMP plugin's `[vimeo]` shortcode callback: https://github.com/ampproject/amp-wp/pull/3678
6. Do `$ npm install && composer install`
7. Activate the AMP plugin
8. Create a new post with a Shortcode block
9. Add a shortcode, like:
```
[vimeo id="84618094" width="600" height="400"]
```
10. Preview the front-end of the AMP URL, like by adding `&amp` or `?amp`to the URL, and look at how the Vimeo shortcode looks
11. `checkout` the `master` branch of Jetpack, not this PR's branch
12. Preview the front-end of the AMP URL, and notice how the Vimeo shortcode looks
13. Expected: The shortcode should look the same, whether using this PR's branch, or the Jetpack plugin's `master` branch
14. Here are some more shortcodes to test:
```
[vimeo id="84618094"]
[vimeo 8461809]
[vimeo id="84618094" width="600"]
[vimeo id="84618094" width="400" height="300"]
[vimeo https://vimeo.com/84618094]
```

## Before

<img width="917" alt="before-vimeo-shortcode" src="https://user-images.githubusercontent.com/4063887/67992159-50aa0f80-fc01-11e9-87b3-3ddcf408f27a.png">

## After

<img width="899" alt="after-vimeo-shortcode" src="https://user-images.githubusercontent.com/4063887/67992178-5bfd3b00-fc01-11e9-94b4-abf0751fe18a.png">

#### Proposed changelog entry for your changes:
Migrate AMP-conversion of Vimeo shortcode from the AMP plugin to Jetpack.
